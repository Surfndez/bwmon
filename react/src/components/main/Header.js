import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import { AppBar, Toolbar, Typography } from '@material-ui/core';

const useStyles = makeStyles({
    title: {
        flexGrow: 1,
    },
});

const Header = () => {
    const classes = useStyles();

    return (
        <AppBar position="static">
            <Toolbar variant="dense">
                <Typography id="app-title" variant="h6" className={classes.title}>
                    {process.env.REACT_APP_DESCRIPTION}
                </Typography>
                <Typography id="app-version" className={classes.text}>
                    {process.env.REACT_APP_VERSION}
                </Typography>
            </Toolbar>
        </AppBar>
    );
};

export default Header;
