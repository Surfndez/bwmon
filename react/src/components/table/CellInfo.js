export default class CellInfo {
    constructor(id, sortable, align, label, footer, convert) {
        this.id = id;
        this.sortable = sortable;
        this.align = align;
        this.label = label;
        this.footer = footer;
        this.convert = convert;
    }
}
