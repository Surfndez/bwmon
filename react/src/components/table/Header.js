import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import TableCell from '@material-ui/core/TableCell';
import TableHead from '@material-ui/core/TableHead';
import TableRow from '@material-ui/core/TableRow';
import TableSortLabel from '@material-ui/core/TableSortLabel';

const useStyles = makeStyles((theme) => ({
    visuallyHidden: {
        border: 0,
        clip: 'rect(0 0 0 0)',
        height: 1,
        margin: -1,
        overflow: 'hidden',
        padding: 0,
        position: 'absolute',
        top: 20,
        width: 1,
    },
}));

const Header = ({ prefix, cellInfos, onRequestSort, ascending, orderBy }) => {
    const classes = useStyles();
    const sortHandler = (property) => (event) => {
        onRequestSort(event, property);
    };

    return (
        <TableHead>
            <TableRow key={'header'}>
                {cellInfos.map((cellInfo) => {
                    if (cellInfo.sortable) {
                        const direction = ascending ? 'asc' : 'desc';
                        return (
                            <TableCell
                                id={prefix + '-' + cellInfo.id}
                                key={cellInfo.id}
                                align={cellInfo.align}
                                sortDirection={orderBy === cellInfo.id ? direction : false}
                            >
                                <TableSortLabel
                                    active={orderBy === cellInfo.id}
                                    direction={orderBy === cellInfo.id ? direction : 'asc'}
                                    onClick={sortHandler(cellInfo.id)}
                                >
                                    {cellInfo.label}
                                    {orderBy === cellInfo.id ? (
                                        <span className={classes.visuallyHidden}>
                                            {direction === 'desc' ? 'sorted descending' : 'sorted ascending'}
                                        </span>
                                    ) : null}
                                </TableSortLabel>
                            </TableCell>
                        );
                    } else {
                        return (
                            <TableCell key={cellInfo.id} align={cellInfo.align}>
                                {cellInfo.label}
                            </TableCell>
                        );
                    }
                })}
            </TableRow>
        </TableHead>
    );
};

export default Header;
