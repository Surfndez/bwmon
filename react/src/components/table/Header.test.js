import React from 'react';
import { create } from 'react-test-renderer';
import Header from './Header';
import CellInfo from './CellInfo';

describe('Header', () => {
    const cellInfos = [
        new CellInfo('IP', true, 'left', 'IP'),
        new CellInfo('MAC', false, 'left', 'MAC'),
        new CellInfo('download', true, 'right', 'Down'),
        new CellInfo('upload', false, 'right', 'Up'),
    ];

    const handleRequestSort = (event, property) => {};

    test('renders correctly', () => {
        const tree = create(
            <Header prefix="test" cellInfos={cellInfos} onRequestSort={handleRequestSort} ascending={true} orderBy={cellInfos[0].id} />
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
