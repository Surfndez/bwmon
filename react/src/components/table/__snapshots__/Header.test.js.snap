// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<thead
  className="MuiTableHead-root"
  role={null}
>
  <tr
    className="MuiTableRow-root MuiTableRow-head"
    role={null}
  >
    <th
      aria-sort="ascending"
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
      id="test-IP"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root MuiTableSortLabel-active"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        IP
        <span
          className="makeStyles-visuallyHidden-1"
        >
          sorted ascending
        </span>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
      scope="col"
    >
      MAC
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      id="test-download"
      scope="col"
    >
      <span
        aria-disabled={false}
        className="MuiButtonBase-root MuiTableSortLabel-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        Down
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionAsc"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
          />
        </svg>
      </span>
    </th>
    <th
      aria-sort={null}
      className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight"
      scope="col"
    >
      Up
    </th>
  </tr>
</thead>
`;
